### PR TITLE
feat(cts): add tests for chunkedBatch wrappers

### PR DESCRIPTION
**Pull request by millotp**
## 🧭 What and Why

🎟 JIRA Ticket: [DI-2506](https://algolia.atlassian.net/browse/DI-2506)

Generate helpers for all the wrapper helpers and assert the requests with a mock server

[DI-2506]: https://algolia.atlassian.net/browse/DI-2506?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: scripts/cts/testServer/replaceAllObjects.ts [hunk 1]
```diff
@@ -74,24 +61,17 @@ function addRoutes(app: Express): void {
...
-          `invalid waitTask count: ${raoState[lang].waitTaskCount}, expected 6`,
-        );
+        expect(raoState).to.include.keys(lang);
+        expect(raoState[lang]).to.deep.equal({
+          copyCount: 2,
+          batchCount: 10,
+          waitTaskCount: 6,
+          tmpIndexName: req.params.indexName,
+          waitingForFinalWaitTask: false,
+          successful: false,
```
Comment (shortcuts) [inline scripts/cts/testServer/replaceAllObjects.ts:71]: next up: JSON based generator for the test server

File: specs/search/helpers/saveObjects.yml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 method:
-  post:
+  get:
```
Comment (shortcuts) [inline specs/search/helpers/saveObjects.yml:2]: it's a post, no?

File: templates/python/search_helpers.mustache [hunk 3]
```diff
@@ -265,7 +265,7 @@
         """
         Helper: Replaces object content of all the given objects according to their respective `objectID` field. The `chunked_batch` helper is used under the hood, which creates a `batch` requests with at most 1000 objects in it.
         """
-        return await self.chunked_batch(index_name=index_name, objects=objects, action=Action.PARTIALUPDATEOBJECT and create_if_not_exists or Action.PARTIALUPDATEOBJECTNOCREATE)
+        return await self.chunked_batch(index_name=index_name, objects=objects, action=Action.PARTIALUPDATEOBJECT if create_if_not_exists else Action.PARTIALUPDATEOBJECTNOCREATE)
```
Comment (shortcuts) [inline templates/python/search_helpers.mustache:268]: possessed by lua

**Review by shortcuts — COMMENTED** [review]
awesome

File: scripts/cts/testServer/replaceAllObjects.ts (hunk 1 above)
Comment (millotp) [reply scripts/cts/testServer/replaceAllObjects.ts:71]: coming soon ™️ !

File: specs/search/helpers/saveObjects.yml (hunk 2 above)
Comment (millotp) [reply specs/search/helpers/saveObjects.yml:2]: its neither, its a helper, there is no method associated with it, but for consistency I put `get` everywhere.
For now it doesn't really matter but if in the script we expect all `helpers` to be `GET` it makes like easier

Comment (millotp) [reply specs/search/helpers/saveObjects.yml:2]: for example for `replaceAllObjects` that call multiple endpoints its not possible to tell what is the method, its only here to comply with redocly

Comment (shortcuts) [reply specs/search/helpers/saveObjects.yml:2]: yup it's just that it seems wrong when reading the specs, we should maybe just add a disclaimer somewhere that those specs are only for generating models (like in the folder root directory)

**Review by shortcuts — APPROVED** [review]
all good for me then, gg for the fixes along the way!

File: specs/search/helpers/saveObjects.yml (hunk 2 above)
Comment (millotp) [reply specs/search/helpers/saveObjects.yml:2]: will do in next PR !

**Pull request merged**